File: TRIQUERY.java
import java.util.*;
import java.io.*;
 
public class TRIQUERY{
 
    public static void main(String[] args)throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int Q = Integer.parseInt(st.nextToken());
        HashMap<Integer,HashMap<Integer,Integer>> points = new HashMap<Integer,HashMap<Integer,Integer>>(N);
 
	HashMap<Integer,Integer> tmp;
 
        for(int i=0; i<N; i++){
            st = new StringTokenizer(br.readLine());
	    int t = Integer.parseInt(st.nextToken());
	    if(points.containsKey(t) == false){
		tmp = new HashMap<Integer,Integer>();
		tmp.put(Integer.parseInt(st.nextToken()),0);
		points.put(t,tmp);
	    }
	    else{
		tmp = points.get(t);
		tmp.put(Integer.parseInt(st.nextToken()),0);
	    }
        }
	int x,y,d;
	int j,k,count,x1,y1,sum;
        for(int i=0; i<Q; i++){
            st = new StringTokenizer(br.readLine());
	    x = Integer.parseInt(st.nextToken());
	    y = Integer.parseInt(st.nextToken());
	    d = Integer.parseInt(st.nextToken());


	    count=0;
	    //	    HashMap<Integer,Integer> tmp;
	    x1 = x+d;
	    y1 = y+d;
	    sum = x1+y;
	    for(k=x;k<=x1;k++){
		if(points.containsKey(k) == true){
		    tmp = points.get(k);
		    for(j=y;j<=y1;j++)
			if(tmp.containsKey(j) == true){
			    if(k+j <= sum)
				count++;
			}
		}
	    }
	    System.out.println(count);
	    //   result(points,x,y,d);
	}
    }
 
 
    //    public static void result(HashMap<Integer,HashMap<Integer,Integer>> hm,int x,int y,int d){
    // }
    
 
 
} 

File: triquery.c
#include <stdio.h>
int points[300000][2];

inline void fastRead(int *a)
{
  register char c=0;
  while (c<33) c=getchar_unlocked();
  *a=0;
  while (c>33)
    {
      *a=*a*10+c-'0';
      c=getchar_unlocked();
    }
}

int main(){
  int n,t,i,x,y,d,x1,y1;
  int p,q,sum,count;
  fastRead(&n);
  fastRead(&t);
  for(i=0;i<n;i++){
    fastRead(&points[i][0]);
    fastRead(&points[i][1]);
  }
  while(t--){
    count = 0;
    fastRead(&x);
    fastRead(&y);
    fastRead(&d);
    x1 = x+d;
    y1 = y+d;
    sum = x+y+d;
    for(i=0;i<n;i++){
      p = points[i][0];
      q = points[i][1];
      if(p>=x && p<=x1 && q>=y && q<=y1 && (p+q)<=sum)
	count++;
    }
    printf("%d\n",count);
  }
  return 0;
}
